### PR TITLE
MSA developer support to call app API

### PR DESCRIPTION
**Pull request by sureshja**

### CODE REVIEW

File: api-reference/beta/api/application-get.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ One of the following permissions is required to call this API. To learn more, in
 |Permission type      | Permissions (from least to most privileged)              |
 |:--------------------|:---------------------------------------------------------|
 |Delegated (work or school account) | Application.Read.All, Directory.Read.All, Application.ReadWrite.All, Directory.ReadWrite.All, Directory.AccessAsUser.All    |
-|Delegated (personal Microsoft account) | Not supported.    |
+|Delegated (personal Microsoft account) | Application.Read.All    |
```
[the remark this answers is not in the document]
Comment (sureshja) [reply api-reference/beta/api/application-get.md:24]: We generally do not list every applicable permission. However, no harm in adding it here.

File: api-reference/beta/api/application-list.md [hunk 2]
```diff
@@ -15,14 +15,17 @@ Namespace: microsoft.graph
 
 Get the list of [applications](../resources/application.md) in this organization.
 
+> NOTE
+> This API would return owned apps if the incoming token is tied to a personal Microsoft account.
```
[the remark this answers is not in the document]
Comment (sureshja) [reply api-reference/beta/api/application-list.md:19]: Rephrased this sentence. 

@davidmu1 / @dkershaw10 - Do you have a better suggestion?

File: api-reference/beta/api/application-list.md [hunk 3]
```diff
@@ -15,14 +15,17 @@ Namespace: microsoft.graph
...
+
 ## Permissions
 
 One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
 
 | Permission type | Permissions (from least to most privileged) |
 |:--------------- |:------------------------------------------- |
 | Delegated (work or school account) | Application.Read.All, Application.ReadWrite.All, Directory.Read.All, Directory.ReadWrite.All, Directory.AccessAsUser.All    |
-| Delegated (personal Microsoft account) | Not supported. |
+| Delegated (personal Microsoft account) | User.Read.All and Application.Read.All, User.Read.All and Application.ReadWrite.All |
```
[the remark this answers is not in the document]
Comment (sureshja) [reply api-reference/beta/api/application-list.md:28]: I like the suggestions. Incorporated the change.

**Comment by CelesteDG** [comment]
@sureshja - Checking in with you. Is this PR ready to merge?

**Comment by sureshja** [comment]
@dkershaw10 Could you take a look this PR?

**Comment by sureshja** [comment]
@davidmu1 / @CelesteDG - I added `uniqueName` as well to this PR. Could you have a look and sign off?

File: api-reference/beta/api/application-list.md [hunk 4]
```diff
@@ -15,14 +15,17 @@ Namespace: microsoft.graph
 
 Get the list of [applications](../resources/application.md) in this organization.
 
+> NOTE
+> When calling this API using tokens issued for a personal Microsoft account, this API would return the apps owned by the personal Microsoft account. There is no notion of organizations exist for personal Microsoft accounts. In order to list applications owned by specific personal Microsoft accounts this API also requires User.Read permission in addition to Application.Read.All or Application.ReadWrite.All.
```
Comment (CelesteDG) [inline api-reference/beta/api/application-list.md:19]:
```suggestion
> When calling this API using tokens issued for a personal Microsoft account, this API will return the apps owned by the personal Microsoft account. The notion of organizations doesn't exist for personal Microsoft accounts. In order to list applications owned by specific personal Microsoft accounts this API also requires User.Read permission in addition to Application.Read.All or Application.ReadWrite.All.
```

File: api-reference/v1.0/resources/application.md [hunk 5]
```diff
@@ -79,6 +79,7 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
...
 | tags |String collection| Custom strings that can be used to categorize and identify the application. Not nullable.|
 | tokenEncryptionKeyId |String|Specifies the keyId of a public key from the keyCredentials collection. When configured, Azure AD encrypts all the tokens it emits by using the key this property points to. The application code that receives the encrypted token must use the matching private key to decrypt the token before it can be used for the signed-in user.|
+| uniqueName | String | The unique identifier for the application that is assigned to an application when created through Azure Resource Manager. Immutable. Read-only. |
```
Comment (CelesteDG) [inline api-reference/v1.0/resources/application.md:82]:
```suggestion
| uniqueName | String | The unique identifier that's assigned to an application when the application is created through Azure Resource Manager. Immutable. Read-only. |
```

File: api-reference/beta/api/application-list.md (hunk 4 above)
Comment (davidmu1) [reply api-reference/beta/api/application-list.md:19]:
```suggestion
> When calling this API using tokens issued for a personal Microsoft account, this API returns the apps owned by the personal Microsoft account. Microsoft accounts do not use organizations. In order to list applications owned by specific personal Microsoft accounts this API requires the User.Read permission, and the Application.Read.All or Application.ReadWrite.All permission.
```

File: api-reference/beta/resources/application.md [hunk 6]
```diff
@@ -59,34 +59,35 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
...
+| Property | Type | Description |
+|:---------------|:--------|:----------|
+| addIns | [addIn](addin.md) collection | Defines custom behavior that a consuming service can use to call an app in specific contexts. For example, applications that can render file streams [may set the addIns property](/onedrive/developer/file-handlers/?view=odsp-graph-online) for its "FileHandler" functionality. This will let services like Office 365 call the application in the context of a document the user is working on. |
+| api | [apiApplication](apiapplication.md) | Specifies settings for an application that implements a web API. |
+| appId | String | The unique identifier for the application that is assigned to an application by Azure AD. Not nullable. Read-only. |
```
Comment (davidmu1) [inline api-reference/beta/resources/application.md:66]:
```suggestion
| appId | String | The unique identifier for the application that is assigned by Azure AD. Not nullable. Read-only. |
```

File: api-reference/beta/resources/application.md [hunk 7]
```diff
@@ -59,34 +59,35 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
...
+| appRoles | [appRole](approle.md) collection | The collection of roles declared by the application. With [app role assignments](approleassignment.md), these roles can be assigned to users, groups, or other applications' service principals. Not nullable. |
```
Comment (davidmu1) [inline api-reference/beta/resources/application.md:67]:
```suggestion
| appRoles | [appRole](approle.md) collection | The collection of roles assigned to the application. With [app role assignments](approleassignment.md), these roles can be assigned to users, groups, or service principals associated with other applications. Not nullable. |
```

File: api-reference/beta/resources/application.md [hunk 8]
```diff
@@ -59,34 +59,35 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
...
+| createdDateTime | DateTimeOffset | The date and time the application was registered. Read-only. |
+| deletedDateTime | DateTimeOffset | The date and time the application was deleted. Read-only. |
+| displayName | String | The display name for the application. |
+| groupMembershipClaims | String | Configures the `groups` claim issued in a user or OAuth 2.0 access token that the application expects. To set this attribute, use one of the following valid string values:<ul><li>`None`</li><li>`SecurityGroup`: For security groups and Azure AD roles</li><li>`All`: This will get all of the security groups, distribution groups, and Azure AD directory roles that the signed-in user is a member of</li></ul> |
```
Comment (davidmu1) [inline api-reference/beta/resources/application.md:71]:
```suggestion
| groupMembershipClaims | String | Configures the `groups` claim issued in a user or OAuth 2.0 access token that the application expects. To set this attribute, use one of the following string values:<ul><li>`None`</li><li>`SecurityGroup`: For security groups and Azure AD roles</li><li>`All`: This gets all of the security groups, distribution groups, and Azure AD directory roles that the signed-in user is a member of.</li></ul> |
```

File: api-reference/beta/resources/application.md [hunk 9]
```diff
@@ -59,34 +59,35 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
...
+| id | String | The unique identifier for the application. Inherited from [directoryObject](directoryobject.md). Key. Not nullable. Read-only. |
+| identifierUris | String collection | The URIs that identify the application within its Azure AD tenant, or within a verified custom domain if the application is multi-tenant. For more information see [Application Objects and Service Principal Objects](https://docs.microsoft.com/azure/active-directory/develop/app-objects-and-service-principals). The *any* operator is required for filter expressions on multi-valued properties. Not nullable. |
```
Comment (davidmu1) [inline api-reference/beta/resources/application.md:73]:
```suggestion
| identifierUris | String collection | The URIs that identify the application within its Azure AD tenant, or within a verified custom domain if the application is multi-tenant. For more information see [Application Objects and Service Principal Objects](https://docs.microsoft.com/azure/active-directory/develop/app-objects-and-service-principals). The `any` operator is required for filter expressions on multi-valued properties. Not nullable. |
```

File: api-reference/beta/resources/application.md [hunk 10]
```diff
@@ -59,34 +59,35 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
...
+| info | [informationalUrl](informationalurl.md) | Basic profile information of the application such as  app's marketing, support, terms of service and privacy statement URLs. The terms of service and privacy statement are surfaced to users through the user consent experience. For more info, see How to: [Add Terms of service and privacy statement for registered Azure AD apps](/azure/active-directory/develop/howto-add-terms-of-service-privacy-statement). |
```
Comment (davidmu1) [inline api-reference/beta/resources/application.md:74]:
```suggestion
| info | [informationalUrl](informationalurl.md) | Basic profile information of the application, such as it's marketing, support, terms of service, and privacy statement URLs. The terms of service and privacy statement are surfaced to users through the user consent experience. For more information, see [How to: Add Terms of service and privacy statement for registered Azure AD apps](/azure/active-directory/develop/howto-add-terms-of-service-privacy-statement). |
```

File: api-reference/beta/resources/application.md [hunk 11]
```diff
@@ -59,34 +59,35 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
...
+| isFallbackPublicClient | Boolean | Specifies the fallback application type as public client, such as an installed application running on a mobile device. The default value is `false` which means the fallback application type is confidential client such as web app. There are certain scenarios where Azure AD cannot determine the client application type (e.g. [ROPC](https://tools.ietf.org/html/rfc6749#section-4.3) flow where it is configured without specifying a redirect URI). In those cases Azure AD will interpret the application type based on the value of this property.|
```
Comment (davidmu1) [inline api-reference/beta/resources/application.md:75]:
```suggestion
| isFallbackPublicClient | Boolean | Specifies the fallback application type as public client, such as an installed application running on a mobile device. The default value is `false` which means the fallback application type is confidential client such as a web app. There are certain scenarios where Azure AD cannot determine the client application type. For example, the [ROPC](https://tools.ietf.org/html/rfc6749#section-4.3) flow where the application is configured without specifying a redirect URI. In those cases Azure AD will interpret the application type based on the value of this property.|
```

File: api-reference/beta/resources/application.md [hunk 12]
```diff
@@ -59,34 +59,35 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
...
+| keyCredentials | [keyCredential](keycredential.md) collection | The collection of key credentials associated with the application Not nullable. |
```
Comment (davidmu1) [inline api-reference/beta/resources/application.md:76]:
```suggestion
| keyCredentials | [keyCredential](keycredential.md) collection | The collection of key credentials associated with the application. Not nullable. |
```

File: api-reference/beta/resources/application.md [hunk 13]
```diff
@@ -59,34 +59,35 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
...
+| logo | Stream | The main logo for the application. Not nullable. |
+| oauth2RequiredPostResponse | Boolean | Specifies whether, as part of OAuth 2.0 token requests, Azure AD will allow POST requests, as opposed to GET requests. The default is false, which specifies that only GET requests will be allowed. |
```
Comment (davidmu1) [inline api-reference/beta/resources/application.md:78]:
```suggestion
| oauth2RequiredPostResponse | Boolean | Specifies whether, as part of OAuth 2.0 token requests, Azure AD allows POST requests, as opposed to GET requests. The default is `false`, which specifies that only GET requests are allowed. |
```

File: api-reference/beta/resources/application.md [hunk 14]
```diff
@@ -59,34 +59,35 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
...
+| onPremisesPublishing |[onPremisesPublishing](onpremisespublishing.md)| Represents the set of properties required for configuring Application Proxy for this application. Configuring these properties allow you to publish your on-premises application for secure remote access. |
```
Comment (davidmu1) [inline api-reference/beta/resources/application.md:79]:
```suggestion
| onPremisesPublishing |[onPremisesPublishing](onpremisespublishing.md)| Represents the set of properties required for [configuring Application Proxy](/graph/application-proxy-configure-api) for this application. Configuring these properties allows you to publish your on-premises application for secure remote access. |
```

File: api-reference/beta/resources/application.md [hunk 15]
```diff
@@ -59,34 +59,35 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
...
+| optionalClaims | [optionalClaims](optionalclaims.md) | Application developers can configure optional claims in their Azure AD apps to specify which claims they want in tokens sent to their application by the Microsoft security token service. See [provide optional claims to your Azure AD app](/azure/active-directory/develop/active-directory-optional-claims) for more information.|
```
Comment (davidmu1) [inline api-reference/beta/resources/application.md:80]:
```suggestion
| optionalClaims | [optionalClaims](optionalclaims.md) | Application developers can configure optional claims in their Azure AD applications to specify the claims that are sent to their application by the Microsoft security token service. For more information, see [How to: Provide optional claims to your app](/azure/active-directory/develop/active-directory-optional-claims).|
```

File: api-reference/beta/resources/application.md [hunk 16]
```diff
@@ -59,34 +59,35 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
...
+| parentalControlSettings | [parentalControlSettings](parentalcontrolsettings.md) |Specifies parental control settings for an application. |
+| passwordCredentials | [passwordCredential](passwordcredential.md) collection|The collection of password credentials associated with the application. Not nullable.|
+| publicClient | [publicClientApplication](publicclientapplication.md) | Specifies settings for installed clients such as desktop or mobile devices. |
+| publisherDomain | String | The verified publisher domain for the application. Read-only.|
+| requiredResourceAccess |[requiredResourceAccess](requiredresourceaccess.md) collection|Specifies resources that this application requires access to and the set of OAuth permission scopes and application roles that it needs under each of those resources. This pre-configuration of required resource access drives the consent experience. Not nullable.|
```
Comment (davidmu1) [inline api-reference/beta/resources/application.md:85]:
```suggestion
| requiredResourceAccess |[requiredResourceAccess](requiredresourceaccess.md) collection| Specifies the resources that the application needs to access. This property also specifies the set of OAuth permission scopes and application roles that it needs for each of those resources. This configuration of access to the required resources drives the consent experience. Not nullable.|
```

File: api-reference/beta/resources/application.md [hunk 17]
```diff
@@ -59,34 +59,35 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
...
+| signInAudience | String | Specifies what Microsoft accounts are supported for the current application. Supported values are:<ul><li>`AzureADMyOrg`: Users with a Microsoft work or school account in my organization’s Azure AD tenant (i.e. single tenant)</li><li>`AzureADMultipleOrgs`: Users with a Microsoft work or school account in any organization’s Azure AD tenant (i.e. multi-tenant)</li> <li>`AzureADandPersonalMicrosoftAccount`: Users with a personal Microsoft account, or a work or school account in any organization’s Azure AD tenant</li></ul> | `AzureADandPersonalMicrosoftAccount` |
```
Comment (davidmu1) [inline api-reference/beta/resources/application.md:86]:
```suggestion
| signInAudience | String | Specifies the Microsoft accounts that are supported for the current application. Supported values are:<ul><li>`AzureADMyOrg`: Users with a Microsoft work or school account in my organization’s Azure AD tenant (single-tenant).</li><li>`AzureADMultipleOrgs`: Users with a Microsoft work or school account in any organization’s Azure AD tenant (multi-tenant)</li><li>`AzureADandPersonalMicrosoftAccount`: Users with a personal Microsoft account, or a work or school account in any organization’s Azure AD tenant.</li></ul> |
```

File: api-reference/v1.0/resources/application.md [hunk 18]
```diff
@@ -79,6 +79,7 @@ This resource supports using [delta query](/graph/delta-query-overview) to track
 | signInAudience | String | Specifies what Microsoft accounts are supported for the current application. Supported values are:<ul><li>`AzureADMyOrg`: Users with a Microsoft work or school account in my organization’s Azure AD tenant (i.e. single tenant)</li><li>`AzureADMultipleOrgs`: Users with a Microsoft work or school account in any organization’s Azure AD tenant (i.e. multi-tenant)</li> <li>`AzureADandPersonalMicrosoftAccount`: Users with a personal Microsoft account, or a work or school account in any organization’s Azure AD tenant</li></ul> |
```
Comment (davidmu1) [inline api-reference/v1.0/resources/application.md:79]:
```suggestion
| signInAudience | String | Specifies the Microsoft accounts that are supported for the current application. Supported values are:<ul><li>`AzureADMyOrg`: Users with a Microsoft work or school account in my organization’s Azure AD tenant (single tenant)</li><li>`AzureADMultipleOrgs`: Users with a Microsoft work or school account in any organization’s Azure AD tenant (multi-tenant)</li><li>`AzureADandPersonalMicrosoftAccount`: Users with a personal Microsoft account, or a work or school account in any organization’s Azure AD tenant.</li></ul> |
```

File: api-reference/v1.0/resources/application.md (hunk 5 above)
Comment (davidmu1) [inline api-reference/v1.0/resources/application.md:82]:
```suggestion
| uniqueName | String | The unique identifier for the application that is assigned to it when it is created through Azure Resource Manager. Immutable. Read-only. |
```

Comment (sureshja) [reply api-reference/v1.0/resources/application.md]: For some reason the tool did not allow me to "Commit suggestion". I manually applied this suggested change.

**Comment by sureshja** [comment]
@CelesteDG @davidmu1 - Thanks you for the suggestions. Sounds like I need to move the `uniqueName` change to a different PR as the code change is not going to be ready until next week. Since MSA changes are live already, I would like to publish the MSA developer support. Will make the change and let you know.

**Comment by sureshja** [comment]
@CelesteDG / @davidmu1 - As mentioned above, I removed `uniqueName` from this PR. Please do the needful to get this published. The /beta/application is live with the MSA developer support today.

**Comment by CelesteDG** [comment]
> @CelesteDG / @davidmu1 - As mentioned above, I removed `uniqueName` from this PR. Please do the needful to get this published. The /beta/application is live with the MSA developer support today.

Thank you, @sureshja - Just waiting for the build to finish then I'll go ahead and merge this PR.

**Comment by Lauragra** [comment]
Please incorporate the comments from David on the PR before we merge; thanks.

**Comment by CelesteDG** [comment]
Hello @sureshja - Please see the latest comments from Laura. Not seeing the feedback from David addressed/incorporated into the latest builds/preview so I removed the 'ready to merge' label for now until those can get incorporated./addressed Thank you.

**Comment by sureshja** [comment]
@CelesteDG, I will look into it and address the pending comments. We also found a new issue in the service which Krill is looking into. Hence this is not ready to be published yet. I will update you back.

**Comment by sureshja** [comment]
@CelesteDG / @Lauragra - The /beta endpoint is updated. This can be published.

**Pull request merged**